### PR TITLE
SNAP-330 Fix unhandled java exceptions

### PR DESCRIPTION
**Pull request by iradaik**
Fixed exceptions described in SNAP-330 story

### CODE REVIEW

File: src/com/sungardas/enhancedsnapshots/tasks/AWSRestoreVolumeTask.java [hunk 1]
```diff
@@ -128,97 +129,117 @@ private void restoreFromSnapshot() {
...
+            String size = backupentry.getSizeGiB();
+            String iops = backupentry.getIops();
+            if (Thread.interrupted()) {
+                throw new EnhancedSnapshotsInterruptedException("Task interrupted");
+            }
+            notificationService.notifyAboutTaskProgress(taskEntry.getId(), "Creating volume...", 15);
+            switch (VolumeType.fromValue(volumeType)) {
+                case Standard:
+                    tempVolume = awsCommunication.createGP2Volume(Integer.parseInt(size));
+                    LOG.info("Created GP2 volume:\n" + tempVolume.toString());
```
Comment (rabay108) [inline src/com/sungardas/enhancedsnapshots/tasks/AWSRestoreVolumeTask.java]: LOG.info Created standard volume

File: src/com/sungardas/init/CredentialsServiceImpl.java [hunk 2]
```diff
@@ -64,10 +65,8 @@
     private final String catalinaHomeEnvPropName = "catalina.home";
     private final String confFolderName = "conf";
     private final String propFileName = "amazon.properties";
-    private final String accessKeyPropName = "amazon.aws.accesskey";
-    private final String secretKeyPropName = "amazon.aws.secretkey";
     private final String DEFAULT_LOGIN = "admin@enhancedsnapshots";
-    private AWSCredentials credentials;
+    private AWSCredentialsProvider credentialsProvider;
     private String instanceId;
     private static final Set<String> regionSet = new HashSet<String>(Arrays.asList(
```
Comment (rabay108) [inline src/com/sungardas/init/CredentialsServiceImpl.java]: why do we need this variable now? as it was just used for the temporary fix to bypass that region issue by me

**Comment by rabay108** [comment]
LGTM

**Comment by kmcgrath** [comment]
@iradaik this request has conflicts with issue #32 (SNAP-312) which was merged with develop first.  The conflicts must be resolved then we can complete the PR.

**Comment by rabay108** [comment]
LGTM

**Pull request merged**